### PR TITLE
Fix/assets improve code and docblocks

### PR DESCRIPTION
**Pull request by cliffordp**
https://central.tri.be/issues/129061

Dug into assets stuff while working this ticket, although this PR doesn't actually address the ticket's issue(s). Just thought it best to separate this stuff out for easier review since there are a lot of little changes.

### CODE REVIEW

File: src/functions/template-tags/general.php [hunk 1]
```diff
@@ -588,43 +588,49 @@ function tribe_register_error( $indexes, $message ) {
...
  *
- * @return string
+ * @param string|array  $group  Which group(s) should be enqueued.
  */
 function tribe_asset_enqueue_group( $group ) {
-	return tribe( 'assets' )->enqueue_group( $group );
+	/** @var Tribe__Assets $assets */
+	$assets = tribe( 'assets' );
+
+	$assets->enqueue_group( $group );
```
Comment (mitogh) [inline src/functions/template-tags/general.php]: Missing a return here.

File: src/functions/template-tags/general.php [hunk 2]
```diff
@@ -588,43 +588,49 @@ function tribe_register_error( $indexes, $message ) {
...
  *
- * @return string
+ * @param string|array $slug Slug to enqueue
  */
 function tribe_asset_enqueue( $slug ) {
-	return tribe( 'assets' )->enqueue( $slug );
+	/** @var Tribe__Assets $assets */
+	$assets = tribe( 'assets' );
+
+	$assets->enqueue( $slug );
```
Comment (mitogh) [inline src/functions/template-tags/general.php]: Missing return here as well.

File: src/Tribe/Settings.php [hunk 3]
```diff
@@ -295,45 +295,51 @@ public function addNetworkPage() {
...
-							admin_url( self::$parent_page )
-						)
-					);
-				}
-				$this->fields_for_save = (array) apply_filters( 'tribe_settings_fields', array() );
-				do_action( 'tribe_settings_after_do_tabs' );
-				$this->fields = (array) apply_filters( 'tribe_settings_fields', array() );
-				$this->validate();
+			if (
+				! isset( $_GET['page'] )
```
Comment (mitogh) [inline src/Tribe/Settings.php]: Personal preference here but I will change `! isset` with a `empty` as removes the need to use a `!` operator

File: src/functions/template-tags/general.php (hunk 1 above)
Comment (cliffordp) [reply src/functions/template-tags/general.php]: enqueue_group() has no `@return` (i.e. returns void) so this shouldn't either, correct?

File: src/functions/template-tags/general.php (hunk 2 above)
Comment (cliffordp) [reply src/functions/template-tags/general.php]: enqueue() has no `@return` / returns void

File: src/Tribe/Settings.php (hunk 3 above)
Comment (cliffordp) [reply src/Tribe/Settings.php]: I like empty(), too. Updated.

File: src/functions/template-tags/general.php (hunk 1 above)
Comment (mitogh) [reply src/functions/template-tags/general.php]: In that case the docblock should be updated accordingly then, as right now it says it returns a string or an array.

File: src/functions/template-tags/general.php (hunk 2 above)
Comment (mitogh) [reply src/functions/template-tags/general.php]: In that case the docblock should be updated accordingly then, as right now it says it returns a string or an array.

Comment (cliffordp) [reply src/functions/template-tags/general.php]: That's what I did. It used to have @return; now it doesn't.

File: src/functions/template-tags/general.php (hunk 1 above)
Comment (cliffordp) [reply src/functions/template-tags/general.php]: Same: That's what I did. It used to have `@return`; now it doesn't.

File: src/functions/template-tags/general.php (hunk 2 above)
Comment (mitogh) [reply src/functions/template-tags/general.php]: My bad I was overlooking at `@param`

Comment (cliffordp) [reply src/functions/template-tags/general.php]: @mitogh - okay to merge?

**Pull request merged**